### PR TITLE
Feature/services in di

### PR DESCRIPTION
**Pull request by piotr-major-rmtcfm-com**

### CODE REVIEW

File: app/controllers/Auth/LoginController.js [hunk 1]
```diff
@@ -40,34 +39,32 @@ class LoginController {
      *        "user": null
      *     }
      */
-    constructor(userRepository) {
+    constructor(userRepository, auth) {
         this.userRepository = userRepository;
+        this.auth = auth;
+        console.log(userRepository);
```
Comment (piotr-sko-prpl) [inline app/controllers/Auth/LoginController.js]: 👀

File: app/services/Auth.js [hunk 2]
```diff
@@ -3,13 +3,13 @@ const jwt = require('jsonwebtoken');
...
 class Auth {
-    static signIn(user) {
+    signIn(user) {
         return jwt.sign({ id: user.getDataValue('id') }, config.jwt.secret, {
             expiresIn: 60 * 60 * 24
         });
     }
 
-    static comparePasswords(password, dbPassword) {
+    comparePasswords(password, dbPassword) {
```
Comment (piotr-sko-prpl) [inline app/services/Auth.js:12]: please use the async version

File: app/controllers/Auth/LoginController.js [hunk 3]
```diff
@@ -40,34 +39,31 @@ class LoginController {
      *        "user": null
      *     }
      */
-    constructor(userRepository) {
+    constructor(userRepository, auth) {
         this.userRepository = userRepository;
+        this.auth = auth;
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/Auth/LoginController.js]: call it `authService`

File: app/controllers/User/StoreController.js [hunk 4]
```diff
@@ -46,14 +44,15 @@ class StoreController {
...
         this.userRepository = userRepository;
+        this.auth = auth;
     }
 
-    async invoke(request, response, next) {
+    async invoke(request, response) {
         const user = await this.userRepository.create(request.body);
 
-        const token = await Auth.signIn(user);
+        const token = await this.auth.signIn(user);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/User/StoreController.js]: authService

File: app/services/Auth.js [hunk 5]
```diff
@@ -3,14 +3,14 @@ const jwt = require('jsonwebtoken');
 const config = require('../../config');
 
 class Auth {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/services/Auth.js]: class AuthService

**Review by daniel-pupczyk-rmtcfm-com — APPROVED** [review]

**Pull request merged**